content/docs/get-started/aws/modify-program.md

```diff
@@ -12,176 +12,159 @@ menu:
 aliases: ["/docs/quickstart/aws/modify-program/"]
 ---
 
-Now that we have an instance of our Pulumi program deployed, let's enable encryption on our S3 bucket.
+Now that your S3 Bucket is provisioned, let's add an object to your bucket. First let's create a new directory called `site`.
 
-Replace the entire contents of {{< langfile >}} with the following:
+```bash
+$ mkdir site
+```
+
+Next let's create an `index.html` file you will upload to your bucket.
+
+{{< chooser os "macos,linux,windows" / >}}
+
+{{% choosable os macos %}}
+
+```bash
+$ touch site/index.html
+```
+
+{{% /choosable %}}
+
+{{% choosable os linux %}}
+
+```bash
+$ touch site/index.html
+```
+
+{{% /choosable %}}
+
+{{% choosable os windows %}}
+
+```bash
+$ type nul > site/index.html
+```
+
+{{% /choosable %}}
+
+Once you've created your `index.html` file, let's add some content:
+
+```html
+<html>
+    <body>
+        <h1>Hello Pulumi</h1>
+    </body>
+</html>
+```
+
+Now that you have your new `index.html` with some content, let's modify your program to upload the file to your S3 Bucket. To successfully upload the file you need to give Pulumi the absolute path to your file. We can easily accomplish this by taking advantage of libraries native to your programming language of choice.
```

Comment:
Another minor linguistic nit, but I'd go with either "let's modify the program" or "modify your program", but not "let's modify your program", because "we" aren't modifying "my" program, I am. 

```suggestion
Now that you have your new `index.html` with some content, let's modify the program to upload the file to your S3 bucket. To successfully upload the file, you need to provide its absolute path. You can easily accomplish this by taking advantage of libraries native to your programming language of choice.
```

.. except that you don't actually have to provide an absolute path, here. You can just give it a relative one:

```
const bucketObject = new aws.s3.BucketObject("index.html", {
    bucket: bucket,
    source: new pulumi.asset.FileAsset("site/index.html"),
});
```

I love showing how easy it it to be able to reach for a library/helper, though.  You could get the same effect using `content` and `fs.readFileSync`:

```
import * as fs from "fs";
...
const bucketObject = new aws.s3.BucketObject("index.html", {
    bucket: bucket,
    content: fs.readFileSync("site/index.html").toString(),
});
```

Also, I wonder whether we should maybe be a bit more careful about using the word "upload" so many times, as it has an imperative ring to it. The code _looks_ like it uploads a file, and new users may read it and think that's what's happening, but of course it doesn't actually do that -- it declares that the file should exist, and this declarative/desired-state thing is something we want to make sure new users don't get the wrong idea about. Know what I mean?